panic running security-secretstore-setup and security-proxy-setup

# 🐞 Bug Report

Running security-secretstore-setup with res/configuration.toml missing causes a panic due to undefined LoggingClient.

See:

```
./security-secretstore-setup --init=true
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x20 pc=0x8a5ae4]

goroutine 17 [running]:
github.com/edgexfoundry/edgex-go/internal/pkg/config.LoadFromFile(0x0, 0x0, 0x0, 0x0, 0x8e1480, 0xc0000b3d40, 0x5d8a5505, 0xc0223daddb)
        /home/ijohnson/git/edgex/edgex-go/internal/pkg/config/loader.go:52 +0x214
github.com/edgexfoundry/edgex-go/internal/security/secretstore.initializeConfiguration(0xbf5ab321623dad00, 0x0, 0x0, 0x0, 0x0, 0xe47d20, 0xe47d01, 0x1)
        /home/ijohnson/git/edgex/edgex-go/internal/security/secretstore/init.go:68 +0x78
github.com/edgexfoundry/edgex-go/internal/security/secretstore.Retry(0xc000140100, 0x0, 0x0, 0x0, 0x0, 0x7530, 0xc00013c050, 0xc0001401e0)
        /home/ijohnson/git/edgex/edgex-go/internal/security/secretstore/init.go:37 +0x131
created by github.com/edgexfoundry/edgex-go/internal/pkg/startup.Bootstrap
        /home/ijohnson/git/edgex/edgex-go/internal/pkg/startup/bootstrap.go:37 +0xe4
```

This is because the LoggingClient in LoadFromFile is nil because we haven't been able to load the configuration file at this point in time yet.

### Affected Services

The issue is located in: security-secretstore-setup


### Is this a regression?

Maybe...

This was from master @ abc9af1de59ab5c4130efe3914aaf2c6688b4e14

--- a/cmd/security-secretstore-setup/main.go
+++ b/cmd/security-secretstore-setup/main.go
@@ -69,6 +69,10 @@ func main() {
 		BootTimeout: internal.BootTimeoutDefault,
 	}
 	startup.Bootstrap(params, secretstore.Retry, logBeforeInit)
+	if secretstore.Configuration == nil {
+		// secretstore.LoggingClient wasn't initialized either
+		os.Exit(1)
+	}
 
 	//step 1: boot up secretstore general steps same as other EdgeX microservice
 

--- a/internal/constants.go
+++ b/internal/constants.go
@@ -14,17 +14,18 @@
 package internal
 
 const (
-	BootTimeoutDefault             = BootTimeoutSecondsDefault * 1000
-	BootTimeoutSecondsDefault      = 30
-	BootRetrySecondsDefault        = 1
-	ClientMonitorDefault           = 15000
-	ConfigFileName                 = "configuration.toml"
-	ConfigRegistryStemCore         = "edgex/core/"
-	ConfigRegistryStemSecurity     = "edgex/security/"
-	ConfigMajorVersion             = "1.0/"
-	LogDurationKey                 = "duration"
-	SecuritySecretsSetupServiceKey = "edgex-security-secrets-setup"
-	SecurityProxySetupServiceKey   = "edgex-security-proxy-setup"
+	BootTimeoutDefault                 = BootTimeoutSecondsDefault * 1000
+	BootTimeoutSecondsDefault          = 30
+	BootRetrySecondsDefault            = 1
+	ClientMonitorDefault               = 15000
+	ConfigFileName                     = "configuration.toml"
+	ConfigRegistryStemCore             = "edgex/core/"
+	ConfigRegistryStemSecurity         = "edgex/security/"
+	ConfigMajorVersion                 = "1.0/"
+	LogDurationKey                     = "duration"
+	SecuritySecretStoreSetupServiceKey = "edgex-security-secretstore-setup"
+	SecuritySecretsSetupServiceKey     = "edgex-security-secrets-setup"
+	SecurityProxySetupServiceKey       = "edgex-security-proxy-setup"
 )
 
 const (

--- a/internal/security/secretstore/init.go
+++ b/internal/security/secretstore/init.go
@@ -22,6 +22,7 @@ import (
 	"github.com/edgexfoundry/edgex-go/internal"
 	"github.com/edgexfoundry/edgex-go/internal/pkg/config"
 	"github.com/edgexfoundry/go-mod-core-contracts/clients/logger"
+	"github.com/edgexfoundry/go-mod-core-contracts/models"
 )
 
 // Global variables
@@ -34,6 +35,11 @@ func Retry(useRegistry bool, configDir, profileDir string, timeout int, wait *sy
 		var err error
 		// When looping, only handle configuration if it hasn't already been set.
 		if Configuration == nil {
+			// Next two lines are workaround for issue #1814 (nil panic while logging)
+			// where config.LoadFromFile depends on a global LoggingClient that isn't set anywhere
+			// Remove this workaround once this tool is migrated to common bootstrap.
+			lc := logger.NewClient(internal.SecuritySecretStoreSetupServiceKey, false, "", models.InfoLog)
+			config.LoggingClient = lc
 			Configuration, err = initializeConfiguration(useRegistry, configDir, profileDir)
 			if err != nil {
 				ch <- err
@@ -46,7 +52,7 @@ func Retry(useRegistry bool, configDir, profileDir string, timeout int, wait *sy
 			} else {
 				// Setup Logging
 				logTarget := setLoggingTarget()
-				LoggingClient = logger.NewClient(internal.SecurityProxySetupServiceKey, Configuration.Logging.EnableRemote, logTarget, Configuration.Writable.LogLevel)
+				LoggingClient = logger.NewClient(internal.SecuritySecretStoreSetupServiceKey, Configuration.Logging.EnableRemote, logTarget, Configuration.Writable.LogLevel)
 			}
 		}
 		// This seems a bit artificial here due to lack of additional service requirements

